File: rot13/rot13.py
#! /usr/bin/env python3

from sys import argv

def rot13(sentence):

    new_sentence = ""
    for letter in sentence:
        
        if 'a' <= letter.lower() <= 'm':
            new_sentence += chr(ord(letter) + 13)
        elif 'n' <= letter.lower() <= 'z':
            new_sentence += chr(ord(letter) - 13)
        else:
            new_sentence += letter

    print(new_sentence)

# No arguments means a loop:
if len(argv) == 1:
    try:
        while 1:
            rot13(input())
    except EOFError: pass
    except KeyboardInterrupt: pass

# Otherwise we'll rot13 the arguments:
else:
    rot13(" ".join(argv[1:]))

""" TAIL INFO:
Name ROT13
Language: Python3
State: Done

Rot13 a string



Example: ./rot13.py hello world
"""

File: rot13/rot13.rb
#! /usr/bin/env ruby

def rot13(sentence)
  puts (sentence.tr("A-Za-z", "N-ZA-Mn-za-m"))
end

# No arguments means a loop:
if ARGV.length == 0 then

  begin
    while 1 do
      rot13(gets.chomp)
    end
  rescue Interrupt
  rescue NoMethodError
  end
  
# Otherwise, we'll rot13 the arguments:
else
  rot13(ARGV.join(" "))
end

=begin TAIL INFO:
Name ROT13
Language: Ruby
State: Done

Rot13 a string



Example: ./rot13.rb hello world
=end
